independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/RequestDeserializeHandler.java

```diff
@@ -25,12 +26,15 @@
     private static final Logger log = Logger.getLogger(RequestDeserializeHandler.class);
 
     private final Class<?> type;
+    private final Type genericType;
     private final MediaType mediaType;
     private final ServerSerialisers serialisers;
     private final int parameterIndex;
 
-    public RequestDeserializeHandler(Class<?> type, MediaType mediaType, ServerSerialisers serialisers, int parameterIndex) {
+    public RequestDeserializeHandler(Class<?> type, Type genericType, MediaType mediaType, ServerSerialisers serialisers,
```

Comment:
Pretty scary that the TCK doesn't test this.

Reply:
BTW, do we need a test for generic input types, or is this part of the next commits with `List<String>` request bodies?

Reply:
You mean something like `List<T>` for the input? If so, then we don't have such a test and we should add it.

Reply:
Well, any generic type, really.

Reply:
Test added in the last commit